front/src/common/api/osrdEditoastApi.ts

```diff
@@ -1488,108 +1473,119 @@ export type ViewMetadata = {
   tiles?: string[];
   type?: string;
 };
+export type LightModeEffortCurves = {
+  is_electric: boolean;
+};
+export type LightEffortCurves = {
+  default_mode: string;
+  modes: {
+    [key: string]: LightModeEffortCurves;
+  };
+};
 export type SpeedDependantPower = {
   powers: number[];
   speeds: number[];
 };
-export type Catenary = {
-  efficiency: number;
-  energy_source_type: 'Catenary';
-  max_input_power: SpeedDependantPower;
-  max_output_power: SpeedDependantPower;
+export type RefillLaw = {
+  soc_ref: number;
+  tau: number;
 };
 export type EnergyStorage = {
   capacity: number;
-  refill_law: {
-    soc_ref: number;
-    tau: number;
-  } | null;
+  refill_law?: RefillLaw | null;
   soc: number;
   soc_max: number;
   soc_min: number;
 };
-export type PowerPack = {
-  efficiency: number;
-  energy_source_type: 'PowerPack';
-  energy_storage: EnergyStorage;
-  max_input_power: SpeedDependantPower;
-  max_output_power: SpeedDependantPower;
-};
-export type Battery = {
-  efficiency: number;
-  energy_source_type: 'Battery';
-  energy_storage: EnergyStorage;
-  max_input_power: SpeedDependantPower;
-  max_output_power: SpeedDependantPower;
-};
 export type EnergySource =
-  | ({
+  | {
+      efficiency: number;
       energy_source_type: 'Catenary';
-    } & Catenary)
-  | ({
+      max_input_power: SpeedDependantPower;
+      max_output_power: SpeedDependantPower;
+    }
+  | {
+      efficiency: number;
       energy_source_type: 'PowerPack';
-    } & PowerPack)
-  | ({
+      energy_storage: EnergyStorage;
+      max_input_power: SpeedDependantPower;
+      max_output_power: SpeedDependantPower;
+    }
+  | {
+      efficiency: number;
       energy_source_type: 'Battery';
-    } & Battery);
-export type RollingStockBase = {
-  base_power_class: string | null;
+      energy_storage: EnergyStorage;
+      max_input_power: SpeedDependantPower;
+      max_output_power: SpeedDependantPower;
+    };
+export type Gamma = {
+  type: string;
+  value: number;
+};
+export type LoadingGaugeType =
+  | 'G1'
+  | 'G2'
+  | 'GA'
+  | 'GB'
+  | 'GB1'
+  | 'GC'
+  | 'FR3.3'
+  | 'FR3.3/GB/G2'
+  | 'GLOTT';
+export type RollingStockMetadata = {
+  detail: string;
+  family: string;
+  grouping: string;
+  number: string;
+  reference: string;
+  series: string;
+  subseries: string;
+  type: string;
+  unit: string;
+};
+export type RollingResistance = {
+  A: number;
+  B: number;
+  C: number;
+  type: string;
+};
+export type LightRollingStock = {
+  base_power_class?: string | null;
```

Comment:
Can this field really be undefined?

Reply:
https://github.com/osrd-project/osrd/blob/dev/editoast/src/schema/rolling_stock/mod.rs#L22